### PR TITLE
Add ActiveRecord 6.0 support

### LINKED ISSUE #175: Rails 6 support
Given I update to rails 6 in my Gemfile:
```diff
diff --git a/Gemfile b/Gemfile
 
-gem 'rails', '~>5.2'
+gem 'rails', '~>6.0.0.beta2'

```

Then I will not be able to update to Rails 6 due to a limit in crypt_keeper. Error message (when running `bundle update rails`:

```
    crypt_keeper (~> 2.0.0.rc2) was resolved to 2.0.1, which depends on
      activesupport (< 5.3, >= 4.2)

    rails (~> 6.0.0.beta2) was resolved to 6.0.0.beta2, which depends on
      activesupport (= 6.0.0.beta2)
```

What are the plans to update this plugin and maintaining the releases?

### PR DESCRIPTION
**Pull request by jesperronn**
Fixes #175

### CODE REVIEW

**Comment by jesperronn** [comment]
@jmazzi I need some help understanding why this fails

**Comment by mtchavez** [comment]
Would be great to get this passing and ready for review. @jesperronn Looks like the postgres service in TravisCI had issues coming up. Did you try rebuilding since those initial CI runs? Maybe it was an intermittent issue.

```bash
$ travis_setup_postgresql 9.3

Starting PostgreSQL v9.3

Assertion failed on job for postgresql@9.3-main.service.

sudo systemctl start postgresql@9.3-main
sql -c 'CREATE DATABASE crypt_keeper_providers;' -U postgres

psql: could not connect to server: No such file or directory

	Is the server running locally and accepting

	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?

The command "psql -c 'CREATE DATABASE crypt_keeper_providers;' -U postgres" failed and exited with 2 during .
```

Also noticed [the docs][docs] quote the postgres version string when using _addons_ instead of _services_
```yaml
addons:
  postgresql: "9.4"
```

[docs]: https://docs.travis-ci.com/user/database-setup/#using-a-different-postgresql-version

**Comment by mtchavez** [comment]
@jesperronn It might be missing the services section to tell postgres to start. Not sure how this is intended to be set up though.
https://docs.travis-ci.com/user/database-setup/#postgresql

**Comment by mtchavez** [comment]
Disregard that. I think you use one or the other addons or services.

**Comment by jesperronn** [comment]
Thanks for helping troubleshoot the build failures, @mtchavez 

I am currently trying out some of your suggestions, starting with quoting the postgres version, updating the version.  Let's see if I can make Travis behave.

You are free to experiment on your own, if you like (create separate pull-request)

**Comment by mtchavez** [comment]
There is an old travis-ci issue suggesting to restart the service in a before script.

```yaml
before_install:
  - sudo service postgresql restart 9.3
```

I also see the use of `sudo: false` which is deprecated per the docs and might be adding to the confusion here. https://docs.travis-ci.com/user/reference/trusty/#container-based-infrastructure

**Comment by jesperronn** [comment]
Please note that the errors are now Rails 6 related for Mysql and Postgres. Examples from latest travis build

```
Failures:
  1) CryptKeeper::LogSubscriber::MysqlAes AES encryption filters aes functions
     Failure/Error: model.method(:delete_all).call
     
     ActiveRecord::StatementInvalid:
       Mysql2::Error: Table 'crypt_keeper_providers.schema_migrations' doesn't exist

```

and 

```
  5) CryptKeeper::LogSubscriber::PostgresPgp Symmetric encryption filters pgp functions
     Failure/Error: model.method(:delete_all).call
     
     ActiveRecord::StatementInvalid:
       PG::UndefinedTable: ERROR:  relation "schema_migrations" does not exist
       LINE 1: DELETE FROM "schema_migrations"
```

Examples from [Travis build 786.17](https://travis-ci.org/jmazzi/crypt_keeper/jobs/578579070)

I REALLY need help fixing this. It seems Rails 6 per default needs some kind of command to maintain test schema or to migrate. But which? 

I previously tried running `rake db:migrate`, as a `before_script` step in travis, but that Rake command does not exist in this project (since it is not a Rails  application)

All help and suggestions are greatly appreciated. 

Also note that I believe that this is only a problem for Travis testing. I don't believe this is a problem for a Rails 6 application using Crypt Keeper.

File: crypt_keeper.gemspec [hunk 1]
```diff
@@ -18,8 +18,8 @@ Gem::Specification.new do |gem|
 
   gem.post_install_message = "WARNING: CryptKeeper 2.0 contains breaking changes and may require you to reencrypt your data! Please view the README at https://github.com/jmazzi/crypt_keeper for more information."
 
-  gem.add_runtime_dependency 'activerecord',  '>= 4.2', '< 5.3'
-  gem.add_runtime_dependency 'activesupport', '>= 4.2', '< 5.3'
+  gem.add_runtime_dependency 'activerecord',  '>= 4.2', '< 6.1'
```
Comment (taylorwilliams) [inline crypt_keeper.gemspec]: Should this be 6.0?

File: crypt_keeper.gemspec [hunk 2]
```diff
@@ -18,8 +18,8 @@ Gem::Specification.new do |gem|
...
+  gem.add_runtime_dependency 'activesupport', '>= 4.2', '< 6.1'
```
Comment (taylorwilliams) [inline crypt_keeper.gemspec]: Ditto

**Comment by kevin-j-m** [comment]
I don't know if this is novel, but only because I see all the references are regarding CI, but I can reproduce this on my local environment.

```
crypt_keeper|rails6 ⇒ rspec ./spec/crypt_keeper/provider/postgres_pgp_spec.rb:23
[Coveralls] Set up the SimpleCov formatter.
[Coveralls] Using SimpleCov's default settings.
Run options: include {:focus=>true, :locations=>{"./spec/crypt_keeper/provider/postgres_pgp_spec.rb"=>[23]}}

Randomized with seed 19061
F

Failures:

  1) CryptKeeper::Provider::PostgresPgp#encrypt Strings should not eq "test"
     Failure/Error: model.method(:delete_all).call

     ActiveRecord::StatementInvalid:
       PG::UndefinedTable: ERROR:  relation "schema_migrations" does not exist
       LINE 1: DELETE FROM "schema_migrations"
                           ^
```

**Comment by kevin-j-m** [comment]
I added a `puts ActiveRecord::Base.descendants` into the [after hook](https://github.com/jmazzi/crypt_keeper/blob/5e282444339e956abddfd05e8c4f4088c3af36a5/spec/spec_helper.rb#L16) that's running on all the tests.

Running on ActiveRecord 5.2:
```ruby
crypt_keeper|rails6⚡ ⇒ bundle exec appraisal activerecord_5_2 rspec ./spec/crypt_keeper/provider/postgres_pgp_spec.rb:23
...
Randomized with seed 51569
ActiveRecord::InternalMetadata
...
```

Running on ActiveRecord 6.0:
```ruby
crypt_keeper|rails6⚡ ⇒ bundle exec appraisal activerecord_6_0 rspec ./spec/crypt_keeper/provider/postgres_pgp_spec.rb:23
...
Randomized with seed 22212
ActiveRecord::InternalMetadata
ActiveRecord::SchemaMigration
primary::SchemaMigration
...
```

So on Rails 6, there are two additional descendants that don't appear in prior versions:
```
ActiveRecord::SchemaMigration
primary::SchemaMigration
```

That obviously tracks with the error message about the `schema_migrations` table not existing, and I'd imagine the `primary` namespace has to do with multiple database support added in Rails 6.

`ActiveRecord::SchemaMigration` has been a descendent of `ActiveRecord::Base` for a [while](https://github.com/rails/rails/blame/6-0-stable/activerecord/lib/active_record/schema_migration.rb), and it doesn't appear that the `DescendantsTracker` [changed](https://github.com/rails/rails/blame/6-0-stable/activesupport/lib/active_support/descendants_tracker.rb) in a way I can easily tell why this may have been introduced. That being said, I'm largely unfamiliar with the source code of rails here in this area, but maybe this will spark a discussion that others can pick up if I'm unable to find anything else?

Otherwise, I apologize for the noise, but if I'm able to dedicate additional time to dig into this and find anything, I'll post my progress/thoughts/findings.

**Comment by kevin-j-m** [comment]
I created #184, which has a proposed solution around the schema_migrations tables, but now has some issues around supporting the proper version(s) of sqlite across the testing matrix of supported ActiveRecord versions.

**Comment by jmazzi** [comment]
Closed by #184

**Pull request closed without merging**